functions/agents/orchestrator.py

```
@@ -0,0 +1,568 @@
+            
+            await self.firestore.update_estimate(
+                estimate_id,
+                {"status": "failed", "error": str(e)}
+            )
+            
+            raise PipelineError(
+                code=ErrorCode.PIPELINE_FAILED,
+                message=f"Pipeline failed: {str(e)}",
+                estimate_id=estimate_id,
+                current_agent=pipeline_status.current_agent
+            )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)